Custom localoptions.h support, official code is available in hg repository but didn't make it yet into any release version.

--- options.h
+++ options.h
@@ -5,6 +5,8 @@
 #ifndef DROPBEAR_OPTIONS_H_
 #define DROPBEAR_OPTIONS_H_
 
+#include "localoptions.h"
+
 /* Define compile-time options below - the "#ifndef DROPBEAR_XXX .... #endif"
  * parts are to allow for commandline -DDROPBEAR_XXX options etc. */
 
@@ -305,7 +307,9 @@
 
 /* This is used by the scp binary when used as a client binary. If you're
  * not using the Dropbear client, you'll need to change it */
+#ifndef DROPBEAR_PATH_SSH_PROGRAM
 #define DROPBEAR_PATH_SSH_PROGRAM "/usr/bin/dbclient"
+#endif
 
 /* Whether to log commands executed by a client. This only logs the 
  * (single) command sent to the server, not what a user did in a 
@@ -347,7 +351,9 @@
 #define DEFAULT_IDLE_TIMEOUT 0
 
 /* The default path. This will often get replaced by the shell */
+#ifndef DEFAULT_PATH
 #define DEFAULT_PATH "/usr/bin:/bin"
+#endif
 
 /* Some other defines (that mostly should be left alone) are defined
  * in sysoptions.h */
